cpp/benchmarks/io/orc/orc_writer_chunks.cpp

```diff
@@ -0,0 +1,123 @@
+                                              num_cols),
+                                 table_size_bytes{data_size});
+  cudf::table_view view = tbl->view();
+
+  auto mem_stats_logger = cudf::memory_stats_logger();
+  cuio_source_sink_pair source_sink(io_type::VOID);
+  for (auto _ : state) {
+    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+    cudf_io::orc_writer_options opts =
+      cudf_io::orc_writer_options::builder(source_sink.make_sink_info(), view);
+    cudf_io::write_orc(opts);
+  }
+
+  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * data_size);
+  state.counters["peak_memory_usage"] = mem_stats_logger.peak_memory_usage();
+  state.counters["encoded_file_size"] = source_sink.size();
+}
+
+void ORC_write_chunked(benchmark::State& state)
+{
+  cudf::size_type num_cols   = state.range(0);
+  cudf::size_type num_tables = state.range(1);
+
+  std::vector<std::unique_ptr<cudf::table>> tables;
+  for (cudf::size_type idx = 0; idx < num_tables; idx++) {
+    tables.push_back(create_random_table(cycle_dtypes({cudf::type_id::INT32,
+                                                       cudf::type_id::FLOAT64,
+                                                       cudf::type_id::STRING,
+                                                       cudf::type_id::DECIMAL32,
+                                                       cudf::type_id::INT8},
+                                                      num_cols),
+                                         table_size_bytes{size_t(data_size / num_tables)}));
+  }
+
+  auto mem_stats_logger = cudf::memory_stats_logger();
+  cuio_source_sink_pair source_sink(io_type::VOID);
+  for (auto _ : state) {
+    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+    cudf_io::chunked_orc_writer_options opts =
+      cudf_io::chunked_orc_writer_options::builder(source_sink.make_sink_info());
+    cudf_io::orc_chunked_writer writer(opts);
+    std::for_each(tables.begin(), tables.end(), [&writer](std::unique_ptr<cudf::table> const& tbl) {
+      writer.write(*tbl);
+    });
+    writer.close();
+  }
+
+  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * data_size);
+  state.counters["peak_memory_usage"] = mem_stats_logger.peak_memory_usage();
+  state.counters["encoded_file_size"] = source_sink.size();
+}
+
+#define OWBM_BENCHMARK_DEFINE(name, size, num_columns)                                 \
+  BENCHMARK_DEFINE_F(OrcWrite, name)(::benchmark::State & state) { ORC_write(state); } \
+  BENCHMARK_REGISTER_F(OrcWrite, name)                                                 \
+    ->Args({num_columns})                                                              \
+    ->Unit(benchmark::kMillisecond)                                                    \
+    ->UseManualTime()
+
+OWBM_BENCHMARK_DEFINE(3Gb8Cols, data_size, 8);
+OWBM_BENCHMARK_DEFINE(3Gb1024Cols, data_size, 1024);
```

Comment:
what do you think about reducing this to 64? All other benchmarks use tables with 64 columns.

Reply:
Completely fine with that. I made it extreme to verify this PR didn't impact the performance.

Reply:
That's fine. I was mostly verifying that this code wasn't a performance hit. I'm fine with reducing that back down to 64.